contracts/AJMarket.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+/// @custom:security-contact security@ajcoin.xyz
+contract AJMarket is Ownable {
+    using Counters for Counters.Counter;
+
+    IERC20 ajCoin;
+
+    // Total count of all listings
+    Counters.Counter private _listingCount;
+
+    // Mapping from listing ID to supply
+    mapping(uint256 => uint256) private _listingSupply;
+
+    // Mapping from listing ID to price
+    mapping(uint256 => uint256) private _listingPrice;
+
+    // Total count of all orders
+    Counters.Counter private _orderCount;
+
+    // Mapping from order ID to listing ID
+    mapping(uint256 => uint256) private _orderListing;
+
+    // Mapping from order ID to user address
+    mapping(uint256 => address) private _orderUser;
+
+    // Mapping from order ID to price
+    mapping(uint256 => uint256) private _orderPrice;
+
+    // Mapping user address to order count
+    mapping(address => uint256) private _orderCountForUser;
+
+    // Mapping from user to list of order IDs
+    mapping(address => mapping(uint256 => uint256)) private _orders;
+
+    event Purchase(address indexed user, uint256 indexed listingId, uint256 price);
+
+    constructor(address _ajCoin) {
+        ajCoin = IERC20(_ajCoin);
```

Comment:
Just curious as to whether the ERC20's are to be minted in the context of this contract. Or if the coins are assumed to be pre-minted before entering into this market contract?

Reply:
no, these coins are already minted in a separate contract and have been "airdropped to many ppl in the practice already! haha

Reply:
ah! makes sense